### ISSUE TITLE
[Open-Meteo] Location search fails

### ISSUE THREAD

**Issue by GitHubQueenn** [breezy-weather/breezy-weather#731]
### Steps to reproduce

Since updating to the current latest version, my home screen widget is stuck on "refreshin" and the app itself will not let me set a manual location. Anytime, I type the location and hit "search," I get, "Location search failed." I am using GrapheneOS and the app worked fine all this past week when I first got it until this update. From F-Droid repo.

### Expected behavior

Location should load.

### Actual behavior

Location not loading.

### Weather source used

Any

### Breezy Weather version

5.0.3-alpha

### Android version

Android 14

### Device

Pixel 6

### Other details

Since updating to the current latest version, my home screen widget is stuck on "refreshin" and the app itself will not let me set a manual location. Anytime, I type the location and hit "search," I get, "Location search failed." I am using GrapheneOS and the app worked fine all this past week when I first got it until this update.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I read the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document and followed instructions if relevant.
- [X] I have updated the app to **[version 5.0.3 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
No changes was done on GPS side in v5.0, try reading #721.

Tl;dr: if you don't use Google Play Services, you may need to first get your location from another app. Then, it should work better with subsequent refreshes.

**Comment by papjul** [comment]
Ok, what happens when you type "Paris" in the location search? There is a known issue on Open-Meteo side that when there is 0 results for a query, it fails instead of saying it found 0 results and you need to try a different query.

**Comment by GitHubQueenn** [comment]
> Ok, what happens when you type "Paris" in the location search? There is a known issue on Open-Meteo side that when there is 0 results for a query (see #270), it fails instead of saying it found 0 results and you need to try a different query.

Well, it's the same location as when it worked fine for me this past week. It randomly disappeared on my home screen widget (the "refreshin" thing I mentioned) so I checked the app and my locations were cleared. Looks like it's actually working if I just type my state in (or your Paris example). But why would a previously-working location randomly disappear and now not be able to be found?

**Comment by papjul** [comment]
> my locations were cleared

Are you talking about pinned issue #705?

> But why would a previously-working location randomly disappear and now not be able to be found?

Maybe you didn’t type the exact same search terms than the previous time you searched for your location? In either case, Open-Meteo provides, by default, the location search results, and we can do nothing about the results, but you can switch to a different location search source like you did. I suggest we close this issue in favor of #270 for the incorrect error message.
